### ISSUE TITLE
flowgraph optimize_cfg Assertion for if-else-expression with if-else-expression as condition

### ISSUE THREAD

**Issue by 15r10nk** [python/cpython#109889]
# Bug report

### Bug description:

```python
a if (1 if b else c) else d
```

output (Python 3.12.0rc3+):
```python
python: Python/flowgraph.c:1598: optimize_cfg: Assertion `no_redundant_nops(g)' failed.
```
tested with the current 3.12 branch (538f505a3744ab1dd29861a859ab81ab65436144)

### CPython versions tested on:

3.12

### Operating systems tested on:

Linux

### Linked PRs
* gh-109899
* gh-109987
* gh-110048

**Comment by AA-Turner** [comment]
cc: @iritkatriel

**Comment by iritkatriel** [comment]
The optimizer removed NOPs which are redundant (don't have important line number info attached). But there is a bug where it misses the case where there is a basic block that contains a single such NOP (and nothing else). 

This is not a correctness issue, it's only an optimization failing. I will make a PR for 3.12 to just remove the assertion. In 3.13 I will fix this optimization.

**Comment by iritkatriel** [comment]
> I will make a PR for 3.12 to just remove the assertion. In 3.13 I will fix this optimization.

I mean remove the assertion in 3.12.0. We can fix in 3.12.1.
